libwayshot/src/lib.rs

```diff
@@ -93,383 +57,373 @@ pub struct CaptureRegion {
     pub height: i32,
 }
 
-/// Supported image encoding formats.
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub enum EncodingFormat {
-    /// Jpeg / jpg encoder.
-    Jpg,
-    /// Png encoder.
-    Png,
-    /// Ppm encoder.
-    Ppm,
-    /// Qoi encoder.
-    Qoi,
+struct IntersectingOutput {
+    output: WlOutput,
+    region: CaptureRegion,
+    transform: Transform,
 }
 
-impl From<EncodingFormat> for image::ImageOutputFormat {
-    fn from(format: EncodingFormat) -> Self {
-        match format {
-            EncodingFormat::Jpg => image::ImageFormat::Jpeg.into(),
-            EncodingFormat::Png => image::ImageFormat::Png.into(),
-            EncodingFormat::Ppm => image::ImageFormat::Pnm.into(),
-            EncodingFormat::Qoi => image::ImageFormat::Qoi.into(),
-        }
-    }
+/// # Example usage
+///
+/// ```
+/// let wayshot_connection = WayshotConnection::new().unwrap();
+/// let image_buffer = wayshot_connection.screenshot_all().unwrap();
+/// ```
+#[derive(Debug)]
+pub struct WayshotConnection {
+    conn: Connection,
+    globals: GlobalList,
 }
 
-impl From<EncodingFormat> for &str {
-    fn from(format: EncodingFormat) -> Self {
-        match format {
-            EncodingFormat::Jpg => "jpg",
-            EncodingFormat::Png => "png",
-            EncodingFormat::Ppm => "ppm",
-            EncodingFormat::Qoi => "qoi",
-        }
+impl WayshotConnection {
+    pub fn new() -> Result<Self> {
```

Comment:
Maybe we should just keep from_connection. While having ::new would be convenient, it is more sensible to keep the API low level and provide more control.

Reply:
`new` is very convenient when the consumer does not care about the connection which applies to the majority of usage and I do not see any downsides to making both methods available.